README_cmake.md

```diff
@@ -1,106 +1,377 @@
+
+### macOS
+
+On macOS, the [Homebrew](https://formulae.brew.sh/cask/cmake#default) cmake
+package is kept up to date. Simply run
 
 ```
-cmake_minimum_required(VERSION 3.16)
-project(HalideExample)
+$ brew update
+$ brew install cmake
+```
 
-set(CMAKE_CXX_STANDARD 11)  # or newer
-set(CMAKE_CXX_STANDARD_REQUIRED YES)
-set(CMAKE_CXX_EXTENSIONS NO)
+to install the newest version of CMake. If your environment prevents you from
+installing Homebrew, the binary release on
+[Kitware's website](https://cmake.org/download/) is also a viable option.
 
-find_package(Halide REQUIRED)
+### Ubuntu Linux
+
+There are a few good ways to install a modern CMake on Ubuntu:
+
+1. If you're on **Ubuntu Linux 20.04** (focal), then simply running
+   `sudo apt install cmake` is sufficient.
+2. If you are on an older Ubuntu release or would like to use a newer CMake,
+   first try installing via the snap store: `snap install cmake`. Be sure you do
+   not already have `cmake` installed via APT. The snap package automatically
+   stays up to date.
+3. For older versions of **Debian, Ubuntu, Mint, and derivatives**, Kitware
+   provides an [APT repository](https://apt.kitware.com/) with up-to-date
+   releases. Note that this is still useful for Ubuntu 20.04 because it will
+   remain up to date.
+
+For other Linux distributions, check with your distribution's package manager or
+use pip as detailed above. Snap packages are also available.
+
+On WSL 1, the snap service is not available; in this case, prefer to use the APT
+repository. On WSL 2, all methods are available.
+
+## Installing dependencies
+
+We generally recommend using a package manager to fetch Halide's dependencies.
+Except where noted, we recommend using vcpkg on Windows, Homebrew on macOS, and
+apt on Ubuntu 20.04 LTS.
+
+Only LLVM and Clang are _absolutely_ required to build Halide. Halide always
+supports three LLVM versions: the current major version, the previous major
+version, and trunk. For most users, we recommend using a binary release of LLVM
+rather than building it yourself.
+
+However, to run all of the tests and apps, an extended set is needed. This
+includes lld, Python 3, libpng, libjpeg, Doxygen, OpenBLAS, ATLAS, and Eigen3.
+While not required to build any part of Halide, we find that Ninja is the best
+backend build tool across all platforms.
+
+Note that CMake has many special variables for overriding the locations of
+packages and executables. Consult the CMake documentation for these. Normally,
+you should prefer to make sure your environment is sound. For instance, if you
+want CMake to use a particular version of Python, create a virtual environment
+and activate it _before_ configuring Halide.
+
+### Windows
+
+We assume you have vcpkg installed at `D:\vcpkg`. Follow the instructions in the
+vcpkg README to install. First install LLVM via vcpkg.
+
+```
+D:\vcpkg> .\vcpkg install llvm[target-all,clang-tools-extra]:x64-windows
```

Comment:
I'd recommend people turn on enable-assertions for llvm, otherwise it just crashes instead of giving an error message when things go weird.